examples/hoc.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { withKnobs, text } from '@storybook/addon-knobs';
+
+import {
+  PlaidLink,
+  PlaidLinkOnExit,
+  PlaidLinkOnEvent,
+  PlaidLinkOnSuccess,
+} from '../src';
+
+const Link = ({ token }) => {
+  const onExit: PlaidLinkOnExit = (error, metadata) => {
+    console.log('onExit', error, metadata);
+  };
+
+  const onEvent: PlaidLinkOnEvent = (eventName, metadata) => {
+    console.log('onEvent', eventName, metadata);
+  };
+
+  const onSuccess: PlaidLinkOnSuccess = (token, metadata) => {
+    console.log('onSuccess', token, metadata);
+  };
+
+  return (
+    <PlaidLink
+      className="CustomButton"
+      style={{ padding: '20px', fontSize: '16px', cursor: 'pointer' }}
+      token={token}
+      onExit={onExit}
+      onSuccess={onSuccess}
+      onEvent={onEvent}
+    >
+      Open Link and connect your bank!
+    </PlaidLink>
+  );
+};
+
+// storybook config
+export const HOC = () => <Link token={text('Link token', '')} />;
```

Comment:
what is HOC.

Reply:
[Higher Order Component](https://reactjs.org/docs/higher-order-components.html). Although idk if this even qualifies as one... it has just been the name of this in this repo for a while